consensus/service/src/byzantine_ledger/mod.rs

```diff
@@ -147,28 +146,34 @@ impl ByzantineLedger {
 
         // Start worker thread
         let thread_handle = {
-            let is_behind = byzantine_ledger.is_behind.clone();
-            let highest_peer_block = byzantine_ledger.highest_peer_block.clone();
-            let broadcaster = broadcaster.clone();
+            let ledger_sync_service = LedgerSyncService::new(
+                ledger.clone(),
+                peer_manager.clone(),
+                ReqwestTransactionsFetcher::new(tx_source_urls, logger.clone()).unwrap(), // Unwrap?
+                logger.clone(),
+            );
+
+            let mut worker = ByzantineLedgerWorker::new(
+                scp_node,
+                ledger,
+                ledger_sync_service,
+                peer_manager,
+                tx_manager,
+                broadcaster.clone(),
+                receiver,
+                send_scp_message,
+                byzantine_ledger.is_behind.clone(),
+                byzantine_ledger.highest_peer_block.clone(),
+                logger,
+            );
             Some(
                 thread::Builder::new()
-                    .name(format!("ByzantineLedger{:?}", node_id))
-                    .spawn(move || {
-                        ByzantineLedgerWorker::start(
```

Comment:
Initially, `start` took a pretty gnarly and confusing set of arguments -- some of which are effectively outputs. For example, `node_id` and quorum_set` are redundant because you can get those from `wrapped_scp_node`. The atomic variables `is_behind` and `highest_peer_block` are "outputs" that get set by the worker.